### ISSUE TITLE
Compiler does not check when using Inject(hilt) in View.

### ISSUE THREAD

**Issue by minwonki** [google/dagger#2055]
While using hilt, I inject in the View, but I did not add @AndroidEntryPoint, but it was not found at compile time and the app died at runtime.
When injecting in View, it needs to be modified so that it can be checked at compile time.

**Comment by bcorso** [comment]
Yeah, I can see how this can be useful. If the Gradle module contains the Hilt processors it's very likely that you would want your Application/Activity/Fragment/View injected with Hilt, and a compile error is better than a NPE at runtime.

On the other hand, we would probably need to have a flag to disable this check since there are apps/libraries that may only be using Hilt in a subset of their Android classes for various reasons.
